var/spack/repos/builtin/packages/egl/package.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import re
+import sys
+
+from spack.package import *
+
+
+class Egl(BundlePackage):
+    """Placeholder for external EGL(OpenGL) libraries from hardware vendors"""
+
+    homepage = "https://www.khronos.org/egl"
+    maintainers("biddisco")
+
+    version("1.5")
+
+    # This should really be when='platform=linux' but can't because of a
+    # current bug in when and how ArchSpecs are constructed
+    if sys.platform.startswith("linux"):
+        provides("gl@4.5")
```

Comment:
If this is the case, then we also need to define the properties for `gl_headers` and `gl_libs` here.

Part of the reason I hadn't finished the EGL work before is because EGL does not actually provide GL, it sits adjacent to OpenGL and can, optionally, create contexts for it. That OpenGL may be provided by the same library providing GLX on linux systems, or it may use a vendored OpenGL library, or something else (like OpenVG or OpenGL|ES) and then binds the API.

I think for now, and simplicity to get this working, assuming that EGL sits next to OpenGL and `provides` headers and libraries is fine, they just need to be found and exposed here.

Reply:
that particular comment was copied from the opengl provider and might not be true/correct. I'll investigate this further, but for now I'm happy to go with what you say and leave things until they need improving for some other system. we've tested this egl support on several machines at CSCS and so far it is good, but they are all linux and checking mac/other OS would be good for the next PR

Reply:
I am sure this works since EGL is almost always going to sit next to an OpenGL, and CMake finds EGL using the Find OpenGL so so long as we don't support other backends this is fine.

However, since this does provide "gl" which can be used without EGL as a package, then It should also say where the Open GL headers and libraries are. Since we are only really using the OpenGL for ParaView I wouldn't expect an issue as ParaView will re-find the OpenGL parts based on the added EGL path. However, from a Spack point of view, we still should be finding and expos the OpenGL aka "gl" components.

Reply:
Work that we've done in the past on EGL which wasn't merged unfortunately. https://github.com/spack/spack/pull/27726. Maybe we can get some ideas from there.